app/controllers/finance/plannings_controller.rb

```diff
@@ -39,13 +39,21 @@ def create
     end
   end
 
-  def current
-    planning = current_user.finance_plannings(params[:currency]).last
-    categories = current_user.finance_categories
+  def show
+    if params[:id] == "current"
+      planning = current_user.finance_plannings(params[:currency]).last
+    else
+      planning = current_user.finance_plannings.find_by(uuid: params[:id])
+    end
 
-    balance = BalanceResume.new(categories, planning.date_start, planning.date_end, params[:currency]).get
-    computed = PlanningCalculator.new(planning).compute
-    render json: computed.merge(balance)
+    if planning
+      categories = current_user.finance_categories
+      balance = BalanceResume.new(categories, planning.date_start, planning.date_end, planning.currency).get
+      computed = PlanningCalculator.new(planning).compute
+      render json: computed.merge(balance)
+    else 
+      render status: 404
+    end
```

Comment:
```suggestion
    else 
      render status: :not_found
    end
```

para melhor legibilidade